### PR TITLE
Create dedicated tests for script.callFunction exceptionDetails

### PR DESCRIPTION
**Pull request by juliandescottes**
And update the existing test for evaluate to test same API surface (ie all primitive types + complex values in both sync and async scenarios)

### CODE REVIEW

File: webdriver/tests/bidi/script/call_function/exception_details.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+                "stackTrace": any_stack_trace,
+                "text": any_string,
+            },
+        },
+        exception.value.result,
+    )
+
+
+@pytest.mark.asyncio
+async def test_reject_complex_value(bidi_session, top_context):
```
Comment (sadym-chromium) [inline webdriver/tests/bidi/script/call_function/exception_details.py:33]: Maybe combine them with primitives together?

File: webdriver/tests/bidi/script/call_function/exception_details.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        ("'2'", {"type": "string", "value": "2"}),
+        ("Number.NaN", {"type": "number", "value": "NaN"}),
+        ("-0", {"type": "number", "value": "-0"}),
+        ("Infinity", {"type": "number", "value": "Infinity"}),
+        ("-Infinity", {"type": "number", "value": "-Infinity"}),
+        ("3", {"type": "number", "value": 3}),
+        ("1.4", {"type": "number", "value": 1.4}),
+        ("true", {"type": "boolean", "value": True}),
+        ("false", {"type": "boolean", "value": False}),
+        ("42n", {"type": "bigint", "value": "42"}),
```
Comment (sadym-chromium) [inline webdriver/tests/bidi/script/call_function/exception_details.py:99]: Why do we check all the primitives types, but only one non-primitive type?

**Review by sadym-chromium — APPROVED** [review]
Makes sense

File: webdriver/tests/bidi/script/call_function/exception_details.py (hunk 2 above)
Comment (juliandescottes) [reply webdriver/tests/bidi/script/call_function/exception_details.py:99]: No strong reason, I can add a few other collections and complex types. I'm not sure how much we want to check all types all the time. Primitives are easy to cover and don't clutter the test too much.

File: webdriver/tests/bidi/script/call_function/exception_details.py (hunk 1 above)
Comment (juliandescottes) [reply webdriver/tests/bidi/script/call_function/exception_details.py:33]: Good point, let's have one test with parameters. I'll reuse the test sets from the result.py tests

**Comment by juliandescottes** [comment]
@sadym-chromium I made a few changes, we now have only one main test function in each file, with many parameters. Let me know if that's ok for you or if you would prefer splitting that a bit.

File: webdriver/tests/bidi/script/evaluate/exception_details.py [hunk 3]
```diff
@@ -53,64 +30,157 @@ async def test_reject_complex_value(bidi_session, top_context):
...
+                ],
+            },
+        ),
+        ("new WeakMap()", {"type": "weakmap", },),
+        ("new WeakSet()", {"type": "weakset", },),
+        ("new Error('SOME_ERROR_TEXT')", {"type": "error"},),
+        # TODO(sadym): add `iterator` test.
+        # TODO(sadym): add `generator` test.
+        # TODO(sadym): add `proxy` test.
+        ("Promise.resolve()", {"type": "promise", },),
```
Comment (sadym-chromium) [inline webdriver/tests/bidi/script/evaluate/exception_details.py:138]: In our implementation, the all the nested promises should be resolved, so the result of this expression will be `undefined`.

**Comment by sadym-chromium** [comment]
> @sadym-chromium I made a few changes, we now have only one main test function in each file, with many parameters. Let me know if that's ok for you or if you would prefer splitting that a bit.

Still LGTM. And it helped to find a bug in our implementation, thanks!

**Comment by juliandescottes** [comment]
Great, thanks for the review. I assume the review comment about `nested promises should be resolved, so the result of this expression will be undefined` was deleted and relates to the implementation issue you mentioned, so I'm going to go ahead and merge. Hope that's ok but let me know if I missed anything :)

**Pull request merged**